RayRedGoose/swapi
# Character api refactor

#### What's this PR do?

This refactors the characterBox.js page and character fetch API into small, clean functions, and moves the fetches into a single, dynamic function called from apiCalls.js.

This is to improve testing and readability.

#### Where should the reviewer start?  
You can just look at apiCalls and CharacterBox.js.  The fakeCharacters file was also updated, but it was just to remove an irrelevant key-value pair so you can ignore it.  The snapshots were updated as well.

#### How should this be manually tested? 

You've already checked it over, but you can pull down and make sure the tests still pass and the app still runs if you'd like.  They are working for me.

#### What are the relevant tickets?

This addresses the following tasks on our Trello board on the Refactor card
- Refacter CharacterBox.js
-Refactor Character Fetch
